packages/core/src/services/evaluations/compiler/constants.ts

```diff
@@ -0,0 +1,20 @@
+import { MessageRole, type AssistantMessage } from '@latitude-data/compiler'
+import type { DocumentLog, ProviderLog } from '$core/browser'
+
+type ParameterFromLogFn = (_: {
+  documentLog: DocumentLog
+  providerLog: ProviderLog
+}) => unknown
+
+export const PARAMETERS_FROM_LOG: Record<string, ParameterFromLogFn> = {
+  messages: ({ providerLog }: { providerLog: ProviderLog }) => {
+    const assistantMessage = {
+      role: MessageRole.assistant,
+      content: providerLog.responseText,
+      toolCalls: providerLog.toolCalls,
+    } as AssistantMessage
+    return [...providerLog.messages, assistantMessage]
+  },
+  response: ({ providerLog }: { providerLog: ProviderLog }) =>
+    providerLog.responseText,
+}
```

Comment:
`PARAMETERS_FROM_LOGS` currently contains only two parameters: `messages` and `response`.

This means that using any other variable name as if it was a parameter will result in an error.
(Of course, the user can still define their own variables within the code and use it later)